fip-6.md

```diff
@@ -0,0 +1,376 @@
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|receiver_public_key|Yes|fio public key, see FIO public key validation rules.|FIO public key which will receive the locked tokens.|
+|votable|Yes|0 is not votable, 1 is votable.|This indicates if the locked amount is votable.|
+|unlockable|Yes|0 is not unlockable, 1 is unlockable.|This indicates if the lock is unlockable.|
+|lock_periods|Yes|JSON Array of periods. See periods below. Min: 1, must total 100% |The locking periods for this token lock.|
+|amount|Yes|amount SUFs| The amount of tokens to lock in SUFs|
+|max_fee|Yes|max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|actor|Yes|FIO account name|FIO account for the signer|
+
+###### Lock Periods
+a set of locking periods is defined for each lock. the locking periods define the duration of the period, which is relative to the creation time of the lock, and a percent to unlock when the duration expires, durations are processed in the order specified with the earliest durations being first and the latest durations being last in the list of durations. Each percent is specified as a decimal percent.
+
+###### Example
+```
+{
+    "receiver_public_key": "FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj",
+    "votable":0,
+    "unlockable":0,
+    "lock_periods": [ {
+                        “duration”: 86400
+                        “percent: 1.2
+                      }, {
+                        “duration”: 172800
+                        “percent: 90.8
+                      }, {
+                        “duration”: 86400
+                        “percent: 8.0
+                      }
+                    ],
+    "actor": "aftyershcu22",
+    "amount": 40000000000000
+    "max_fee": 40000000000
+}
+```
+##### Processing
+* require auth of the actor
+* Verify the receiver public key.
+* Verify the votable value.
+* Verify the lock periods
+      Minimum of 1
+      check that total is 100%. 
+      check that duration is greater than 0
+* Create account if account does not yet exist for the receiver public key
+*  Verify the locking account has necessary balance.
+* verify that the fee for this does not exceed the max fee specified.
+* charge appropriate fee (this will be a bundled fee transaction, fee will be based on number of periods specified)
+* create entry in the locktokens table for these lock tokens.
+* verify tx does not exceed max transaction size.
+* increase RAM limit by 200+(60*number of lock periods) bytes
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid receiver public key|Invalid format|400|"receiver_public_key"|value of public key|"Invalid Public Key"|
+|Invalid votable|Invalid format|400|"votable"|votable invalid|"Invalid votable value"|
```

Comment:
The 5th value should say "Value sent in, e.g. "100" not "fio public key invalid". You dont want the error to look like this:
{
   "type": "invalid_input",
   "message": "An invalid request was sent in, please check the nested errors for details.",
   "fields": [   {
      "name": "receiver_public_key",
      "value": "fio public key invalid",
      "error": "Invalid Public Key"
   }]
}

Reply:
ill update the fip